tests/data_generator.py

```diff
@@ -110,7 +122,39 @@ def __get_raw_data() -> tuple[NodeFeatures, ArticleFeatures, AllEdges]:
             t.tensor([4.0, 4.1, 4.2, 4.3, 4.4]),
             t.tensor([5.0, 5.1, 5.2, 5.3, 5.4]),
         ]
-    ).type(t.long)
+    )
     graph_edges = t.tensor([[0, 0, 0, 1, 1, 2, 2], [0, 2, 4, 1, 5, 3, 0]]).type(t.long)
 
     return node_features, article_features, graph_edges
+
+
+def __generated(
+    num_users: int = 3,
+    num_user_features: int = 2,
+    num_articles: int = 10,
+    num_article_features: int = 5,
+    connection_ratio: float = 0.5,
+) -> tuple[NodeFeatures, ArticleFeatures, AllEdges]:
+    node_features = t.stack(
```

Comment:
I think this could be separate function, we could reuse it across both :)

Reply:
Across both what? We are reusing it across both!

Reply:
that this can be a util function, so you don't have to copy the implementation detail :)